src/components/FeatureExportButton/FeatureExportButton.js

```diff
@@ -0,0 +1,223 @@
+      for (let i = 0; i < f.getProperties(); i += 1) {
+        if (/^(geometry|name|description)$/.test(i)) {
+          // eslint-disable-next-line no-continue
+          continue;
+        }
+        clone.unset(i, true);
+      }
+
+      let styles;
+
+      if (clone.getStyleFunction()) {
+        styles = clone.getStyleFunction().call(clone);
+      } else {
+        styles = olLayer.getStyleFunction().call(olLayer, clone);
+      }
+      const newStyle = {
+        fill: styles[0].getFill(),
+        stroke: styles[0].getStroke(),
+        text: styles[0].getText(),
+        image: styles[0].getImage(),
+        zIndex: styles[0].getZIndex(),
+      };
+
+      if (newStyle.image instanceof Circle) {
+        newStyle.image = null;
+      }
+
+      if (newStyle.image) {
+        const imgSource = newStyle.image.getSrc();
+        if (!/(http(s?)):\/\//gi.test(imgSource)) {
+          // eslint-disable-next-line no-console
+          console.log(
+            "Local image source isn't support for KML export." +
+              'Should use remote web server',
+          );
+        }
+      }
+
+      // If only text is displayed we must specify an
+      // image style with scale=0
+      if (newStyle.text && !newStyle.image) {
+        newStyle.image = new Icon({
+          src: 'noimage',
+          scale: 0,
+        });
+      }
+
+      const myStyle = new Style(newStyle);
+      clone.setStyle(myStyle);
+      exportFeatures.push(clone);
+    });
+
+    if (exportFeatures.length > 0) {
+      if (exportFeatures.length === 1) {
+        // force the add of a <Document> node
+        exportFeatures.push(new Feature());
+      }
+
+      // eslint-disable-next-line new-cap
+      featString = new format({
+        extractStyles: true,
+        defaultStyle: [kmlStyle],
+      }).writeFeatures(exportFeatures);
+
+      // Remove no image hack
+      featString = featString.replace(
+        /<Icon>\s*<href>noimage<\/href>\s*<\/Icon>/g,
+        '',
+      );
+
+      // Remove empty placemark added to have
+      // <Document> tag
+      featString = featString.replace(/<Placemark\/>/g, '');
+
+      // Add KML document name
+      if (layer.name) {
+        featString = featString.replace(
+          /<Document>/,
+          `<Document><name>${layer.name}</name>`,
+        );
+      }
+    }
+
+    return featString;
+  }
+
+  exportFeature() {
+    const { layer } = this.props;
+    const now = new Date()
+      .toJSON()
+      .slice(0, 20)
+      .replace(/[.:T-]+/g, '');
+    const featString = this.createFeatureString(layer);
+
+    const formatString = featString
+      ? featString.match(/<(\w+)\s+\w+.*?>/)[1]
+      : null;
+
+    const fileName = `exported_features_${now}.${formatString || 'xml'}`;
+    const charset = document.characterSet || 'UTF-8';
+    const type = `${
+      formatString === 'kml'
+        ? 'data:application/vnd.google-earth.kml+xml'
+        : 'data:text/plain'
```

Comment:
If the default extension is xml maybe 'text/xml' makes more sense.

Reply:
Indeed :+1: